tests/cpp17/generated_cpp17/monster_test_generated.h

```diff
@@ -2332,6 +2336,176 @@ inline flatbuffers::Offset<Referrable> CreateReferrable(flatbuffers::FlatBufferB
+  testarrayoftables.clear();
+  testarrayoftables.reserve(o.testarrayoftables.size());
+  for (const auto& v : o.testarrayoftables) { testarrayoftables.emplace_back((v) ? new MyGame::Example::MonsterT(*v) : nullptr); }
+  testnestedflatbuffer.clear();
+  testnestedflatbuffer.reserve(o.testnestedflatbuffer.size());
+  for (const auto& v : o.testnestedflatbuffer) { testnestedflatbuffer.emplace_back(v); }
+  testarrayofbools.clear();
+  testarrayofbools.reserve(o.testarrayofbools.size());
+  for (const auto& v : o.testarrayofbools) { testarrayofbools.emplace_back(v); }
+  testarrayofstring2.clear();
+  testarrayofstring2.reserve(o.testarrayofstring2.size());
+  for (const auto& v : o.testarrayofstring2) { testarrayofstring2.emplace_back(v); }
+  testarrayofsortedstruct.clear();
+  testarrayofsortedstruct.reserve(o.testarrayofsortedstruct.size());
+  for (const auto& v : o.testarrayofsortedstruct) { testarrayofsortedstruct.emplace_back(v); }
+  flex.clear();
+  flex.reserve(o.flex.size());
+  for (const auto& v : o.flex) { flex.emplace_back(v); }
+  test5.clear();
+  test5.reserve(o.test5.size());
+  for (const auto& v : o.test5) { test5.emplace_back(v); }
+  vector_of_longs.clear();
+  vector_of_longs.reserve(o.vector_of_longs.size());
+  for (const auto& v : o.vector_of_longs) { vector_of_longs.emplace_back(v); }
+  vector_of_doubles.clear();
+  vector_of_doubles.reserve(o.vector_of_doubles.size());
+  for (const auto& v : o.vector_of_doubles) { vector_of_doubles.emplace_back(v); }
+  vector_of_referrables.clear();
+  vector_of_referrables.reserve(o.vector_of_referrables.size());
+  for (const auto& v : o.vector_of_referrables) { vector_of_referrables.emplace_back((v) ? new MyGame::Example::ReferrableT(*v) : nullptr); }
+  vector_of_weak_references.clear();
+  vector_of_weak_references.reserve(o.vector_of_weak_references.size());
+  for (const auto& v : o.vector_of_weak_references) { vector_of_weak_references.emplace_back(v); }
+  vector_of_strong_referrables.clear();
+  vector_of_strong_referrables.reserve(o.vector_of_strong_referrables.size());
+  for (const auto& v : o.vector_of_strong_referrables) { vector_of_strong_referrables.emplace_back((v) ? new MyGame::Example::ReferrableT(*v) : nullptr); }
+  vector_of_co_owning_references.clear();
+  vector_of_co_owning_references.reserve(o.vector_of_co_owning_references.size());
+  for (const auto& v : o.vector_of_co_owning_references) { vector_of_co_owning_references.emplace_back((v) ? new ReferrableT(*v) : nullptr); }
+  vector_of_non_owning_references.clear();
+  vector_of_non_owning_references.reserve(o.vector_of_non_owning_references.size());
+  for (const auto& v : o.vector_of_non_owning_references) { vector_of_non_owning_references.emplace_back(v); }
+  vector_of_enums.clear();
+  vector_of_enums.reserve(o.vector_of_enums.size());
+  for (const auto& v : o.vector_of_enums) { vector_of_enums.emplace_back(v); }
+}
+
+inline MonsterT &MonsterT::operator=(const MonsterT& o) {
```

Comment:
can one be implemented in terms of the other to save on code size?

Reply:
Possibly yes. I could implement copy using the [copy-and-swap idiom](https://en.cppreference.com/w/cpp/language/operators#Assignment_operator), but it has a performance cost (memory high water mark mostly), and I again optimized for efficiency of the generated code instead of code size or generator complexity. It would also require implementing a swap function, which will generate just as much code as the copy assignment operator.

I could refactor the assignments into a common function invoked from the copy constructor and the copy assignment operator, but this will yield worse copy constructor performance because all members will be initialized twice.

I do wonder if I could do something like

```c++
MonsterT &operator=(const MonsterT& o) {
    MonsterT tmp(o);
    *this = std::move(tmp);
    return *this;
}
```

I haven't studied such an implementation in enough details to know if this would be problematic. This is not a common idiom, at least not one I am familiar with.